oauth2_http/java/com/google/auth/oauth2/IdTokenProvider.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019, Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.auth.oauth2;
+
+import java.io.IOException;
+import java.util.List;
+
+/** Interface for an Google OIDC token Provider. This type represents a google issued OIDC token. */
+public interface IdTokenProvider {
+
+  /**
+   * Enum of various credential-specific options to apply to the token.
+   * <li><b>ComputeEngineCredentials</b>:
+   * <li><em>FORMAT_FULL</em>
+   * <li><em>LICENSES_TRUE</em>
+   * <li><b>ImpersonatedCredential</b>:
+   * <li><em>INCLUDE_EMAIL</em>
+   * <li>
+   */
+  public enum Option {
+    FORMAT_FULL("formatFull"),
+    LICENSES_TRUE("licensesTrue"),
+    INCLUDE_EMAIL("includeEmail");
+
+    private String option;
+
+    Option(String option) {
```

Comment:
private?

Reply:
not sure i follow, it should be  private yeah?  its just the internal artificat (or should i just rename that field though)

Reply:
Yes, unless there's something I'm missing the constructor could be private.

Reply:
oh, ok, i misinterpreted the comment to be for the variable (not constructor).  marked private